app/src/main/java/com/example/cocktaildictionary/viewmodel/HomeActivityViewModel.kt

```diff
@@ -27,18 +33,50 @@ import java.time.LocalDateTime
 
 class HomeActivityViewModel(application: Application): AndroidViewModel(application) {
 
+   companion object{
+      private const val DONE = 0L
+      private const val ONE_SECOND = 1000L
+      private const val COUNTDOWN_TIME = 60000L
+   }
    private var myCompositeDisposable: CompositeDisposable = CompositeDisposable()
+   private val timer : CountDownTimer
    private var cacheDatabase: CacheDatabase
    private val converter:Converters = Converters()
-
+   private lateinit var cocktailList: CocktailList
```

Comment:
Why did you add this var back?

Reply:
The reason I brought this var back is because I use it in the loadFilteredData method. Its purpose is to serve as the cocktail list that will be traversed and checked if it contains the query. I tried to use the cocktail list that is stored cache however since the cache is cleared after 1 min it results in an error when a search query is provided when the cache is empty. I also tried to use the cocktail list from the viewState given that the viewState type is HomeActivityStates.DataLoadSuccessful however with the way my code is structured this approach results in the filtered list being displayed even after the search has been completed and the user wishes to go to the original home screen which displays all the cocktails. I also thought of going with the approach of making a get request for the cocktails and use that cocktail list however this would result in numerous unnecessary requests. Thus I opted to make use of the variable cocktail list.

Reply:
There should be a way to filter without using a backup property. We can explore that in a later stage. For this simple app it will suffice for now.